iceoryx_posh/test/moduletests/test_mepoo_chunk_settings.cpp

```diff
@@ -0,0 +1,350 @@
+                            PayloadParams{0U, PayloadParams::MAX_ALIGNMENT},
+                            PayloadParams{1U, PayloadParams::MAX_ALIGNMENT},
+                            PayloadParams{sizeof(PayloadOffset_t), PayloadParams::MAX_ALIGNMENT},
+                            PayloadParams{sizeof(PayloadOffset_t) * 42U, PayloadParams::MAX_ALIGNMENT}));
+
+uint32_t expectedChunkSizeWithCustomHeader(const PayloadParams& payload, uint32_t customHeaderSize)
+{
+    const uint32_t customHeaderSizeAndPaddingToBackOffset =
+        iox::algorithm::max(customHeaderSize, static_cast<uint32_t>(alignof(PayloadOffset_t)));
+
+    if (payload.alignment <= alignof(PayloadOffset_t))
+    {
+        // back-offset is always adjacent to the custom header (as much as possible with the alignment constraints)
+        constexpr uint32_t BACK_OFFSET_SIZE{sizeof(PayloadOffset_t)};
+        return static_cast<uint32_t>(sizeof(ChunkHeader)) + customHeaderSizeAndPaddingToBackOffset + BACK_OFFSET_SIZE
+               + payload.size;
+    }
+    else
+    {
+        // back-offset is not necessarily adjacent to the custom header
+        const uint32_t paddingBytesAndBackOffsetSize = payload.alignment;
+        return static_cast<uint32_t>(sizeof(ChunkHeader)) + customHeaderSizeAndPaddingToBackOffset
+               + paddingBytesAndBackOffsetSize + payload.size;
+    }
+}
+
+TEST_P(ChunkSettings_AlteringPayloadWithCustomHeader,
+       MultipleCustomHeaderSizesAndAlignments_ResultsIn_RequiredChunkSizeIsCorrect)
+{
+    const auto payload = GetParam();
+
+    SCOPED_TRACE(std::string("Payload size = ") + std::to_string(payload.size) + std::string("; alignment = ")
+                 + std::to_string(payload.alignment));
+
+    constexpr uint32_t CUSTOM_HEADER_SIZES[]{
+        1U, sizeof(ChunkHeader) / 2U, sizeof(ChunkHeader), sizeof(ChunkHeader) * 2U};
+    constexpr uint32_t CUSTOM_HEADER_ALIGNMENTS[]{0U, 1U, alignof(ChunkHeader), alignof(ChunkHeader)};
+
+    for (const auto customHeaderAlignment : CUSTOM_HEADER_ALIGNMENTS)
+    {
+        SCOPED_TRACE(std::string("Custom header alignment = ") + std::to_string(customHeaderAlignment));
+
+        for (const auto customHeaderSize : CUSTOM_HEADER_SIZES)
+        {
+            SCOPED_TRACE(std::string("Custom header size = ") + std::to_string(customHeaderSize));
+
+            if (customHeaderSize < customHeaderAlignment)
+            {
+                // the size must always be a multiple of the alignment
+                continue;
+            }
+
+            const uint32_t expectedSize = expectedChunkSizeWithCustomHeader(payload, customHeaderSize);
+
+            auto sutResult =
+                ChunkSettings::create(payload.size, payload.alignment, customHeaderSize, customHeaderAlignment);
+            ASSERT_FALSE(sutResult.has_error());
+            auto& sut = sutResult.value();
+
+            EXPECT_THAT(sut.requiredChunkSize(), Eq(expectedSize));
+        }
+    }
+}
```

Comment:
We need one more tests which calls `ChunkSettings::customHeaderAlignment`. Maybe you could add this to an already existing test.

Reply:
done